homeassistant/components/motionblinds_ble/config_flow.py

```diff
@@ -68,8 +68,9 @@ async def async_step_bluetooth(
         self._discovery_info = discovery_info
         self._mac_code = get_mac_from_local_name(discovery_info.name)
         self._display_name = DISPLAY_NAME.format(mac_code=self._mac_code)
-        self.context["local_name"] = discovery_info.name
-        self.context["title_placeholders"] = {"name": self._display_name}
+        if self._display_name:
+            # If we have a name, use it as flow title
+            self.context["title_placeholders"] = {"name": str(self._display_name)}
```

Comment:
I'm not sure this is the correct fix here.

I suggest instead to drop __init__ method and mark all as required as they always all get set:
```python
class FlowHandler(ConfigFlow, domain=DOMAIN):
    """Handle a config flow for Motionblinds Bluetooth."""

    _discovery_info: BluetoothServiceInfoBleak | BLEDevice
    _mac_code: str
    _display_name: str
    _blind_type: MotionBlindType
```

Then you can simply drop `local_name`.

Reply:
If you want to keep this refactor for a later PR, then there is still no need for the condition as it is always set here:
```suggestion
        self.context["title_placeholders"] = {"name": self._display_name}
```

Reply:
The problem was that mypy did not narrow down the type of `self._display_name`. Solved by storing the output of the format in a local variable as well as in the instance attribute.

I think replacing instance attributes which are always there in favor of type annotations is a bit error prone, I'd rather not do that in this PR.